src/lua/httpc.lua

```diff
@@ -29,8 +29,7 @@
 --  SUCH DAMAGE.
 --
 
-local driver = package.loaded.http.client
-package.loaded.http = nil
+local driver = require('http.client')
```

Comment:
I would prefer to no, because:

1. If we'll look on it as is, without taking into account further commits, this `package.loaded` entry is replaced with another table in `tarantool_lua_init()` (see line 874):

   https://github.com/tarantool/tarantool/blob/ed1362557f9dba58a7bc781ae8ca2a0dcd5dd85b/src/lua/init.c#L862-L882
2. I'm eliminating direct `package.loaded` maniputations here and there and will go into this direction in further patches.
3. If we'll take the *[4/6] lua: don't use public module name as internal one* commit into account, it replaces the name of the internal module with `http.client.lib` and it is done for many other modules without removing from `package.loaded`. (Moreover, I eliminate such tweak in *[5/6] lua: don't tweak package.loaded.compat* for another module.) I see no sense to add code purely to complicate access to internals: it is anyway possible to access them.

I've added a note to the commit message:

> Dropping of the `package.loaded['http.client']` field in unnecessary,
> because it is anyway replaced by the resulting module table in
> `tarantool_lua_init()`

It is a bit excessive, for my taste, but it is harmless, so let it be.

Reply:
Thanks!